### PR TITLE
Create proc_creation_macos_add_to_admin_group.yml

### PR DESCRIPTION
**Pull request by D4rkCiph3r**
### Summary of the Pull Request
New rule for adding an account new/existing to the admin group, thus granting admin privileges.

### Detailed Description of the Pull Request / Additional Comments
1. Using sysadminctl, it can be used to create and add to the admin group at one go.
2. Using dscl, if there is an existing user, it can be added to the admin group by modifying the "GroupMembership" attribute.
Refer: https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1078.003/T1078.003.md

### Example Log Event
NA (It is very rare for administration team to use custom scripts which uses these utilities to create and add an account to the admin group.

### Fixed Issues
NA

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

**Review by phantinuss — APPROVED** [review]
the rule looks good to me. Maybe we can change the naming from 'suspicious' to something less urgent like 'potentially suspicious'? @nasbench ?

In this rule we alert on the addition to the group and no additional 'suspicious' behaviour. The adding itself is the suspicious/interesting item.

File: rules/macos/process_creation/proc_creation_macos_add_to_admin_group.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+title: Suspicious Inclusion To Admin Group
```
Comment (nasbench) [inline rules/macos/process_creation/proc_creation_macos_add_to_admin_group.yml:1]:
```suggestion
title: User Added To Admin Group - MacOS
```

File: rules/macos/process_creation/proc_creation_macos_add_to_admin_group.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+title: Suspicious Inclusion To Admin Group 
+id: 0c1ffcf9-efa9-436e-ab68-23a9496ebf5b
+status: experimental
+description: Detects suspicious attempts to create and/or add an account to the admin group, thus granting admin privileges
```
Comment (nasbench) [inline rules/macos/process_creation/proc_creation_macos_add_to_admin_group.yml:4]:
```suggestion
description: Detects attempts to create and/or add an account to the admin group, thus granting admin privileges.
```

File: rules/macos/process_creation/proc_creation_macos_add_to_admin_group.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            - ' -admin '
+    selection_dscl: #adds to admin group
+        Image|endswith: '/dscl'
+        CommandLine|contains|all:
+            - ' -append '
+            - ' /Groups/admin '
+            - ' GroupMembership '
+    condition: 1 of selection_*
+falsepositives:
+    - Legitimate administration activities (very rare)
```
Comment (nasbench) [inline rules/macos/process_creation/proc_creation_macos_add_to_admin_group.yml:32]:
```suggestion
    - Legitimate administration activities
```

**Review by nasbench — APPROVED** [review]

**Comment by nasbench** [comment]
> the rule looks good to me. Maybe we can change the naming from 'suspicious' to something less urgent like 'potentially suspicious'? @nasbench ?
> 
> In this rule we alert on the addition to the group and no additional 'suspicious' behaviour. The adding itself is the suspicious/interesting item.

@phantinuss I don't think we can use the word suspicious. Since adding a user to an admin group is a legitimate activity that can be "suspicious" depending on other factors. But the rule is not describing a suspicious act. So I simplified the title.

Let me know what you think.

**Review by phantinuss — APPROVED** [review]

**Pull request merged**